### ISSUE TITLE
Nested routers

### ISSUE THREAD

**Issue by lpww** [yoshuawuyts/wayfarer#63]
I noticed this issue while playing around with the examples/nested.js file. when nesting routers with multiple routes, the last route of a parent router is not added to the trie.

example:

```
var wayfarer = require('wayfarer')

var userRouter = wayfarer('err')
var repoRouter = wayfarer()

repoRouter.on('/:repo', ({ repo, user }) => {
  console.log('repo', repo)
  console.log('user', user)
})
userRouter.on('/user/:user', repoRouter)
```

userRouter has multiple routers, ['user', ':user'], but only ['user'] is added to the trie. After mounting, the full route in the tree is ['user', ':repo']. I would expect the full user route to be included, ['user', ':user', ':repo'].

these are the results of running the above:

```
userRouter('/user/timmy/linklocal')
// Error: route '/user/timmy/linklocal' did not match

userRouter('user/linklocal')
// LOG: repo linklocal
// LOG: user undefined
```

a workaround is to add a trailing slash to the end of the userRoute. '/user/:user/'. it results in the empty string route being removed, leaving the desired route intact. 

possible fix

after looking into the issue, I noticed that trie.mount method is removing parts of the route. it is creating a head when mounting a sub-trie with multiple routes [here](https://github.com/yoshuawuyts/wayfarer/blob/master/trie.js#L118:L121). this head is created as a concat of all but the last element in the route. eg for '/user/:user', the head would be 'user'. the last route, ':user', is never added to the trie.

I'm not sure what is the intention of removing the last route but changing this code to simply pass the [full route](https://github.com/yoshuawuyts/wayfarer/blob/master/trie.js#L118:L121) seems to resolve the issue without breaking any tests. I feel like maybe there might be an untested case, though? I was also able to see a commented out test passing, so wondering if this is a known issue? I'm happy to make a pr for this if it's the right solution

**Comment by marcbachmann** [comment]
@yoshuawuyts do you have any input regarding the uncommented test? Did you hit a different issue except that it failed? It would be nice to get that fixed in nanorouter. I'll also need subroutes in there.

<sub>btw. sorry for linking the other pr to this one. they aren't related at all.</sub>

**Comment by marcbachmann** [comment]
That really fixes it. Thanks
I've added another test for the new `.match` method and created a pr: https://github.com/yoshuawuyts/wayfarer/pull/65

```diff
diff --git a/trie.js b/trie.js
index 2dd4acf..c63103f 100644
--- a/trie.js
+++ b/trie.js
@@ -115,8 +115,7 @@ Trie.prototype.mount = function (route, trie) {
     key = split[0]
     node = this.create(key)
   } else {
-    var headArr = split.splice(0, split.length - 1)
-    var head = headArr.join('/')
+    var head = split.join('/')
     key = split[0]
     node = this.create(head)
   }
```

**Comment by yoshuawuyts** [comment]
This should be fixed by `v6.6.3` :grin:
